File: Java_Backend/E-Tour/src/main/java/entity/Subcategory.java
package entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Subcategory {
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer subcategoryId;
    private String subcategoryName;
    private String subcategoryImagePath;
    private Boolean flag;
    
    @ManyToOne
    @JoinColumn(name="category_id",nullable = false)
    private Category category;

}

File: DotNet_Backend/SafarnamaApplication/SafarnamaApplication/Models/Subcategory.cs
using System;
using System.Collections.Generic;

namespace SafarnamaApplication.Models;

public partial class Subcategory
{
    public int SubcategoryId { get; set; }

    public string SubcategoryCode { get; set; } = null!;

    public int? CategoryId { get; set; }

    public string? SubcategoryName { get; set; }

    public string? SubcategoryImagePath { get; set; }

    public bool? Flag { get; set; }

    public virtual Category? Category { get; set; }

    public virtual ICollection<Package> Packages { get; set; } = new List<Package>();
}
